backend/backend/process_client_json.py

```diff
@@ -0,0 +1,48 @@
+"""Module providing functionality to retrieve requests, parse JSON, and call on
+   roll die functionality.
+
+"""
+from cgitb import enable
+from cgi import FieldStorage
+import json
+from backend.roll_die import roll_two_dice
+
+enable()
+
+SAMPLE_DICTIONARY_OBJECT = {"key": "value"}
+
+
+def manager_function():
+    """Function to act as start point for client requests sending JSON.
+
+    Returns:
+        An ack string to inform the function terminated successfully.
+
+    """
+    print('Content-Type: text/html')
+    print()
+    client_request = interpret_request()
+    dice_result = roll_two_dice()
+    print((client_request, dice_result))
+    return "ack"
+
+
+def interpret_request():
+    """Function to take in parameters sent in client request and extract JSON.
+
+    Returns:
+        A dictionary representing the parsed JSON from the client.
+
+    """
+    print('Content-Type: text/html')
+    print()
+
+    form_data = FieldStorage()
+
+    json_data = form_data.getfirst("jsonrequest")
+    if json_data is None:
```

Comment:
I'm not a fan of serving dummy test data in the production code—see if you can instead stick in the test data in the testing file, so you can mock the FieldStorage. If you can't, you can just run the `json.loads(json_data)` (even though `json_data` will be `None`), and have the unit test expect an exception with [`assertRaises`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) (the exception should be : `TypeError: the JSON object must be str, not 'NoneType'`)

Reply:
I'm trying out the assertRaises approach but it seems to not be catching the TypeError

Reply:
Never mind. I wasn't asserting exceptions in the right places